openquakeplatform/openquakeplatform/static/js/irv_viewer_d3_tree.js

```diff
@@ -279,41 +371,33 @@
 
             nodeUpdate.select("circle")
                 .attr("r", function (d) {
-                    if (d.weight <= 0.10) {
-                        return 2;
-                    }
-                    else if (d.weight > 0.10 && d.weight <= 0.20 ) {
-                        return 4;
-                    }
-                    else if (d.weight > 0.20 && d.weight <= 0.30 ) {
-                        return 6;
-                    }
-                    else if (d.weight > 0.30 && d.weight <= 0.40 ) {
-                        return 8;
-                    }
-                    else if (d.weight > 0.40 && d.weight <= 0.50 ) {
-                        return 10;
-                    }
-                    else if (d.weight > 0.50 && d.weight <= 0.60 ) {
-                        return 12;
-                    }
-                    else if (d.weight > 0.60 && d.weight <= 0.70 ) {
-                        return 14;
-                    }
-                    else if (d.weight > 0.70 && d.weight <= 0.80 ) {
-                        return 16;
-                    }
-                    else if (d.weight > 0.80 && d.weight <= 0.90 ) {
-                        return 18;
-                    }
-                    else if (d.weight > 0.90 && d.weight <= 100 ) {
-                        return 20;
+                    // d.weight is expected to be between 0 and 1
+                    // Nodes are displayed as circles of size between 1 and CIRCLE_SCALE
+                    return d.weight ? Math.max(Math.abs(d.weight) * CIRCLE_SCALE, MIN_CIRCLE_SIZE): MIN_CIRCLE_SIZE;
+                })
+                .style("stroke", function(d) {
+                    if (d.weight < 0) {
+                        return "PowderBlue";
+                    } else {
+                        return "RoyalBlue";
                     }
                 })
+                .style("stroke-width", function(d) {
+                    return d.weight ? Math.min(Math.abs(d.weight) * CIRCLE_SCALE / 2, MAX_STROKE_SIZE): 4;
+                })
                 .style("fill", function(d) {
-                    return d.source ? d.source.linkColor: d.linkColor;
+                    // return d.source ? d.source.linkColor: d.linkColor;
+                    if (d.parent !== undefined && d.parent.operator.indexOf("ignore weights") > -1) {
+                        return "Gold";
```

Comment:
Instead of coloring in Gold nodes which weights are ignored, in the plugin we currently render such nodes with equal radius and we don't render the actual weights